parm/metplus/separate_fcst_obs_info.py

```diff
@@ -0,0 +1,249 @@
+
+    Arguments:
+      args:  
+    """
+
+    # Set up logging.
+    # If the name/path of a log file has been specified in the command line arguments,
+    # place the logging output in it (existing log files of the same name are overwritten).
+    # Otherwise, direct the output to the screen.
+    log_level = str.upper('debug')
+    #FORMAT = "[%(levelname)s:%(name)s:  %(filename)s, line %(lineno)s: %(funcName)s()] %(message)s"
+    #if args.log_fp:
+    #    logging.basicConfig(level=log_level, format=FORMAT, filename=args.log_fp, filemode='w')
+    #else:
+    #    logging.basicConfig(level=log_level, format=FORMAT)
+
+    metplus_conf_dir = Path(os.path.join(home_dir, 'parm', 'metplus')).resolve()
+    print(f'==>> metplus_conf_dir = {metplus_conf_dir}')
+    config_fn = ''.join(['vx_config_', det_or_ens, '.yaml'])
+    config_fp = Path(os.path.join(metplus_conf_dir, config_fn)).resolve()
+    config_dict = load_config_file(config_fp)
+
+    #logging.debug(f"""\nconfig_dict =\n{get_pprint_str(config_dict, '  ')}\n""")
+    print(f"""\nconfig_dict =\n{get_pprint_str(config_dict, '  ')}\n""")
+    #print(f'config_dict = {config_dict}')
+    #laskjdfl
+
+    #print(f'field_group = {field_group}')
+    #fields_levels_threshes_both = config_dict[field_group]
+    fgs_fields_levels_threshes_both = config_dict
+
+    # Set the character that, in the various strings representing a field,
+    # level, or threshold, separates the forecast value from the observation
+    # value.
+    sep_char = '|'
+
+    # Separate the information in the given field name, levels, and thresholds  
+    # dictionary into a fields-and-levels-and-thresholds dictionary for forecasts
+    # and another one for thresholds.
+
+    # The following have to be lists of dictionaries instead of dictionaries
+    # of dictionaries because the field names could be identical and thus 
+    # cannot be used as dictionary keys at the top level.
+    #fgs_fields_levels_threshes_fcst = []
+    #fgs_fields_levels_threshes_obs = []
+    fgs_fields_levels_threshes_fcst = {}
+    fgs_fields_levels_threshes_obs = {}
+    for field_group, fields_levels_threshes_both in fgs_fields_levels_threshes_both.items():
+
+        # Loop over the field names associated with the current field group.
+        #fields_levels_threshes_fcst = {}
+        #fields_levels_threshes_obs = {}
+        fields_levels_threshes_fcst = []
+        fields_levels_threshes_obs = []
+        for field_both, levels_threshes_both in fields_levels_threshes_both.items():
+    
+            # Parse the current field name to extract the forecast field name and the
+            # observation field name.
+            tmp = field_both.split('|')
+            num_sep_chars = len(tmp) - 1
+            field_fcst = tmp[0].strip()
+            if num_sep_chars == 0:
+                field_obs = field_fcst
+            elif num_sep_chars == 1:
+                field_obs = tmp[1].strip()
+            else:
+                error_msg
```

Comment:
I think this logic would be better to avoid "temporary" variables:

```suggestion
            if '|' in field_both:
                if field_both.count('|') == 1:
                    field_fcst, field_obs = field_both.split('|')
                else:
                    error_msg
            else:
                field_obs = field_fcst
```